apis/shared-dimensions/lib/domain/shared-dimensions.ts

```diff
@@ -45,9 +45,7 @@ export function getSharedTerms({ sharedDimension, freetextQuery, validThrough, l
     `
 
   if (freetextQuery) {
-    select = select.WHERE`FILTER (
-      REGEX(${name}, "^${freetextQuery}", "i")
-    )`
+    select = textSearch(select, term, schema.name, freetextQuery)
```

Comment:
I didn't want to couple the implementation of `textSearch` with the variables bound in the query itself, in case it would make sense to re-use it somewhere else. But I have no idea what are the implications of matching the `schema:name` again a second time...

I'll extract the filter if you think it's better.